akka-docs/src/main/paradox/typed/fault-tolerance.md

```diff
@@ -122,6 +122,18 @@ Java
 That means that the `setup` block will only be run when the parent actor is first started, and not when it is
 restarted.
 
+## The PreRestart signal
+
+Before a supervised actor is restart it is sent the @apidoc[akka.actor.typed.PreRestart] signal giving it a chance to clean up resources
+that are created when it is restarted, much like the @apidoc[akka.actor.typed.PostStop] signal when the actor stops. 
```

Comment:
the resources are created when when it is started and after restart in new instance, but maybe keep it simple
"clean up resources that it has created"

Reply:
One thing that occurred to me while doing this PR was that you probably want to do the same thing for PostStop and PreRestart in most cases, would it make sense to introduce a common supertype for those two so one can catch both as one signal?